### DISCUSSION TITLE
If the Router received a Child Update Request from a Device that is not its Child, should the Router send Child Update Response? Must or May?

### DISCUSSION THREAD

**Discussion by jinpeng1989** [openthread/openthread#8531] [Q&A]
The Router received a Child Update Request from a Device, but it do not send Child Update Response. Is this as expected? Check the Specification 1.3. It seems that even when the Device is not its Child, the Parent should send Child Update Response.

![image](https://user-images.githubusercontent.com/97089879/207840330-b9351877-b18f-492e-98f9-5d7b2f9af095.png)

![image](https://user-images.githubusercontent.com/97089879/207839565-cca6a0af-dcb4-4a6a-a80f-4baf99d93305.png)

**Comment by martinzi** [comment]
A router receiving a Child Update Request from a device not in its Child Table **MAY** send a response comprising an error code, e.g. 
![image](https://user-images.githubusercontent.com/30142883/207907780-ee5783ca-67cb-475f-8071-b7250ef2279d.png)

**Reply by jwhui** [reply]
LIne 24 in the copied spec text clearly states "MAY".

OpenThread will only send a Child Update Response to a unknown child if the child indicates that it is rx-on-when-idle:
https://github.com/openthread/openthread/blob/6d87e1a28bca6a7d2cb4eb58cab403b5210b1269/src/core/thread/mle_router.cpp#L2510-L2521

**Reply by jinpeng1989** [reply]
@martinzi @jwhui Thank you. The Child I mentioned above attach to the Router with the Rx-off-when-idle in the Mode TLV. The Timeout for keep-alive message is 900 seconds. The last Data Request was at 08:03:09 (packet No.33351). At 08:05:30 (packet No.33434), the Parent do not reply the Child Update Request from the Child. It seems that the device is not considered to be the child of the Router before timeout.

![image](https://user-images.githubusercontent.com/97089879/207995708-24adbc69-89d7-4d1a-8fa3-b1df6286a729.png)

The Child device address: (wpan.addr16 == 0x485e) or (ipv6.addr eq fe80::7ccc:dd2e:3873:49d)
![image](https://user-images.githubusercontent.com/97089879/207997369-fb817e33-7d80-464d-bf9d-0f49536b4bc9.png)

The attachment is Wireshark capture packets, Network Key is 0166ae8c1f2f104e7d698e836ea665d6


[20221208-temperature_not_renew.zip](https://github.com/openthread/openthread/files/10241815/20221208-temperature_not_renew.zip)

**Reply by jwhui** [reply]
Because the child is rx-off-when-idle, the child must send Data Request messages to retrieve packets from the parent. I do not see Data Request messages being sent after the Child Update Request. My guess is the Parent has a Child Update Response queued for the child, but the child is not retrieving them properly.

**Reply by jinpeng1989** [reply]
Thank you for your support. I got it. There should be Data Request from Child after Child Update Request, then the Child Update Response can be retrieved. 
![image](https://user-images.githubusercontent.com/97089879/208003207-7eeef6d2-3123-409f-bffe-27d74a1335af.png)

**Comment by jinpeng1989** [comment]
This Child device is periodically reattach since the Router do not reply before timeout. What causes this?
![image](https://user-images.githubusercontent.com/97089879/207998747-ee27b29c-492d-439a-b42e-ccea97ef748a.png)

**Reply by jwhui** [reply]
BTW, it would help if you can keep the messages in the same thread, rather than new "suggest an answer" submissions.

**Reply by jinpeng1989** [reply]
OK.
